src/java.net.http/share/classes/jdk/internal/net/http/AsyncSSLConnection.java

```diff
@@ -42,12 +42,13 @@ class AsyncSSLConnection extends AbstractAsyncSSLConnection {
     final PlainHttpPublisher writePublisher;
     private volatile SSLTube flow;
 
-    AsyncSSLConnection(InetSocketAddress addr,
+    AsyncSSLConnection(Origin originServer,
+                       InetSocketAddress addr,
                        HttpClientImpl client,
                        String[] alpn,
                        String label) {
-        super(addr, client, Utils.getServerName(addr), addr.getPort(), alpn, label);
-        plainConnection = new PlainHttpConnection(addr, client, label);
+        super(originServer, addr, client, Utils.getServerName(addr), addr.getPort(), alpn, label);
```

Comment:
Do we still need Utils.getServerName(addr), addr.getPort(), now that we have originServer? Or in other words - should we base these calls on the `addr` parameter or on the `originServer` parameters?